### PR TITLE
adding a cloud build to every travis run

### LINKED ISSUE #656: Automatically run cloud tests
We don't run the cloud tests on every check-in because they take too long, but we should be running them automatically on some schedule so we catch bugs early rather than late.

### PR DESCRIPTION
**Pull request by lbergelson**
we now have a 1+3 matrix which is as follows:

required to pass:
non-cloud non-bucket tests on google dataflow

not required to pass:
non cloud non-bucket tests on spark
cloud and bucket tests on google
cloud and bucket tests on spark

moving TERM out of matrix
setting up service account with guide from https://github.com/GoogleCloudPlatform/appengine-try-python-webapp2/blob/master/.travis.yml
adding install_gcloud.sh copied from a google repo
fixing gs:path
adding key for bucket tests

resolves #656 

Once the google cloud tests are all passing we should move them to the required to pass section

### CODE REVIEW

File: install_gcloud.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+#Install gcloud
+if [ ! -d $HOME/gcloud/google-cloud-sdk ]; then
```
Comment (lbergelson) [inline install_gcloud.sh]: So this script is copied from a google repo.  Not sure what the legality of that is.  Can we have a google person bless it?

File: build.gradle [hunk 2]
```diff
@@ -155,9 +155,13 @@ jar {
 }
 
 test {
-    // enable TestNG support (default is JUnit)
+    String CI = "$System.env.CI"
     useTestNG{
-        excludeGroups 'cloud', 'bucket'
+        if ( CI == "true" &&  "$System.env.CLOUD" =="true"){
```
Comment (droazen) [inline build.gradle]: Can we make it so that to turn on cloud/bucket tests, all you have to do is set CLOUD to true?

File: install_gcloud.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+    mkdir -p $HOME/gcloud &&
+    wget https://dl.google.com/dl/cloudsdk/release/google-cloud-sdk.tar.gz --directory-prefix=$HOME/gcloud &&
+    cd $HOME/gcloud &&
+    tar xzf google-cloud-sdk.tar.gz &&
+    printf '\ny\n\ny\ny\n' | ./google-cloud-sdk/install.sh &&
+    cd $TRAVIS_BUILD_DIR;
+fi
```
Comment (droazen) [inline install_gcloud.sh]: Should the encrypted key really be in the hellbender root?

File: src/main/java/org/broadinstitute/hellbender/engine/dataflow/DataflowCommandLineProgram.java [hunk 4]
```diff
@@ -71,11 +70,11 @@ public static String getRunnerTypeName(String runnerSimpleName) {
...
     @Argument(doc = "path to the client secrets file for google cloud authentication. Ignored if apiKey is set.",
-            shortName = "secret", fullName = "client_secret", optional=true)
-    protected File clientSecret = new File("client_secret.json");
+            shortName = "secret", fullName = "client_secret", optional=true, mutex={"apiKey"})
+    protected File clientSecret;
 
     @Argument(doc = "API Key for Google Cloud (unnecessary if a client secrets file is provided).",
-            shortName = "apiKey", fullName = "apiKey", optional=true)
+            shortName = "apiKey", fullName = "apiKey", optional=true, mutex={"client_secret"})
     protected String apiKey = null;
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/dataflow/DataflowCommandLineProgram.java]: Can we create a ticket to make dataflow authentication easier (eg., automatically populate the API key from the appropriate environment variable)?

File: src/test/java/org/broadinstitute/hellbender/tools/walkers/bqsr/BaseRecalibratorDataflowIntegrationTest.java [hunk 5]
```diff
@@ -139,7 +140,7 @@ public void runAllTests() throws Exception {
 
         return new Object[][]{
                 // reference in cloud, computation local.
-                {new BQSRTest(hg18ReferenceCloud, HiSeqBam, dbSNPb37, "-knownSites " + moreSites, getResourceDir() + "expected.NA12878.chr17_69k_70k.2inputs.txt")},
+                {new BQSRTest(hg18ReferenceCloud, HiSeqBam, dbSNPb37, apiArgs +"-knownSites " + moreSites, getResourceDir() + "expected.NA12878.chr17_69k_70k.2inputs.txt")},
                 // input in cloud, computation local.
                 {new BQSRTest(hg18Reference, HiSeqBamCloud, dbSNPb37, "", getResourceDir() + "expected.NA12878.chr17_69k_70k.txt")},
```
Comment (droazen) [inline src/test/java/org/broadinstitute/hellbender/tools/walkers/bqsr/BaseRecalibratorDataflowIntegrationTest.java]: Specify API key in the 2nd bucket test case as well?

**Comment by droazen** [comment]
Looks good -- feel free to merge after addressing comments.

**Comment by droazen** [comment]
(and consulting with a Google person about the licensing issue)

**Comment by droazen** [comment]
(and fixing the codeship failure)

**Comment by droazen** [comment]
Can we merge this ASAP? Need it to fix https://github.com/broadinstitute/hellbender/issues/683

File: src/main/java/org/broadinstitute/hellbender/engine/dataflow/DataflowCommandLineProgram.java (hunk 4 above)
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/engine/dataflow/DataflowCommandLineProgram.java]: created #701

File: install_gcloud.sh (hunk 1 above)
Comment (lbergelson) [inline install_gcloud.sh]: @wbrockman Could you give us a :+1: on using these several lines of google code?

**Comment by lbergelson** [comment]
@droazen Rebased and waiting for tests.

**Comment by lbergelson** [comment]
@droazen Refactored.  Any additional comments or are we ok?

**Comment by lbergelson** [comment]
Oh, that last comment was on the wrong pull request...

**Comment by wbrockman** [comment]
The install script is nothing to worry about, Louis - it's just following the instructions for getting started with gcloud.

**Pull request merged**